### DISCUSSION TITLE
Should cbor_encode_cred_ext() encode the minPinLength value?

### DISCUSSION THREAD

**Discussion by ntwerdochlib** [Yubico/libfido2#677] [Q&A] [answered]
I see that the corresponding decode_cred_extension() does look for and decode a value for minPinLength, but cbor_encode_cred_ext() only encodes a bool for the extension.

Is this by design or an oversight?

**Comment by LDVG** [comment, marked as answer]
The implementation follows the CTAP2.1 specification, see [§12.4. Minimum PIN Length Extension (minPinLength)](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#sctn-minpinlength-extension):

> **Authenticator extension input**
>
> Boolean asking for minimum PIN length value in Unicode code points. The platform sends the authenticatorMakeCredential request with the following CBOR map entry in the "extensions" field to the authenticator:
>
> ```
> "minPinLength": true
> ```

and

> **Authenticator extension output**
>
> If the RP is
> 
> * authorized, the authenticator sets the minPinLength return value to the current minimum PIN length value.
> * not authorized, the authenticator ignores the extension and does not return any authenticator extension output.
> 
> ```
> CDDL:
> "minPinLength": uint
> ```

i.e. `cbor_encode_cred_ext()` encodes the authenticator extension input; and `decode_cred_extension()` decodes the authenticator extension output.

If your intention is to set a minimum PIN length, you use the [authenticatorConfig](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#authenticatorConfig) command’s [setMinPINLength](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#setMinPINLength) sub-command (see also [§7.4. Set Minimum PIN Length](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#sctn-feature-descriptions-minPinLength)). This is abstracted in libfido2 via the [`fido_dev_set_pin_minlen()`](https://developers.yubico.com/libfido2/Manuals/fido_dev_set_pin_minlen.html) function.

**Reply by ntwerdochlib** [reply]
Well damn.  I saw this extension as a means for the RP to enforce a minimum pin length, not as a request for minimum pin length.  This clears that up for me.

Thanks
